symfony/client/src/orangehrmPimPlugin/pages/employee/CustomField.vue

```diff
@@ -0,0 +1,213 @@
+import {navigate} from '@orangehrm/core/util/helper/navigation';
+import {APIService} from '@orangehrm/core/util/services/api.service';
+
+export default {
+  data() {
+    return {
+      headers: [
+        {
+          name: 'fieldName',
+          slot: 'title',
+          title: 'Custom Field Name',
+          style: {flex: 2},
+        },
+        {name: 'screen', title: 'Screen', style: {flex: 2}},
+        {name: 'fieldType', title: 'Field Type', style: {flex: 2}},
+        {
+          name: 'actions',
+          title: 'Actions',
+          slot: 'action',
+          style: {flex: 1},
+          cellType: 'oxd-table-cell-actions',
+          cellConfig: {
+            delete: {
+              onClick: this.onClickDelete,
+              component: 'oxd-icon-button',
+              props: {
+                name: 'trash',
+              },
+            },
+            edit: {
+              onClick: this.onClickEdit,
+              props: {
+                name: 'pencil-fill',
+              },
+            },
+          },
+        },
+      ],
+      checkedItems: [],
+    };
+  },
+
+  components: {
+    'delete-confirmation': DeleteConfirmationDialog,
+  },
+
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      '/api/v2/pim/custom-fields',
+    );
+    const {
+      showPaginator,
+      currentPage,
+      total,
+      pages,
+      pageSize,
+      response,
+      isLoading,
+      execQuery,
+    } = usePaginate(http);
+
+    return {
+      http,
+      showPaginator,
+      currentPage,
+      isLoading,
+      total: total,
+      pages,
+      pageSize,
+      execQuery,
+      items: response,
+    };
+  },
+  methods: {
+    onClickAdd() {
+      navigate('/pim/saveCustomFields');
+    },
+    onClickEdit(item) {
+      navigate('/pim/saveCustomFields/{id}', {id: item.id});
+    },
+    onClickDeleteSelected() {
+      const ids = this.checkedItems.map(index => {
+        return this.items?.data[index].id;
+      });
+      this.$refs.deleteDialog.showDialog().then(confirmation => {
+        if (confirmation === 'ok') {
+          this.deleteItems(ids);
+        }
+      });
+    },
+    onClickDelete(item) {
+      this.$refs.deleteDialog.showDialog().then(confirmation => {
+        if (confirmation === 'ok') {
+          this.deleteItems([item.id]);
+        }
+      });
+    },
+    deleteItems(items) {
+      if (items instanceof Array) {
+        this.isLoading = true;
+        this.http
+          .deleteAll({
+            ids: items,
+          })
+          .then(() => {
+            return this.$toast.deleteSuccess();
+          })
+          .then(() => {
+            this.isLoading = false;
+            this.resetDataTable();
+          });
+      }
+    },
+    async resetDataTable() {
+      this.checkedItems = [];
+      await this.execQuery();
+    },
+  },
+
+  computed: {
+    updateItemsDataWithFieldType() {
```

Comment:
Please use `usePaginate` for `normalizing` the data. avoid using custom patterns. Refer [PIM Employee list](https://github.com/orangehrm/orangehrm/blob/5.0/symfony/client/src/orangehrmPimPlugin/pages/employee/Employee.vue#L127)